hometasks/7-algorithms-and-data-structures/src/classes.ts

```
@@ -0,0 +1,134 @@
+import { Type } from 'typescript';
+import { VertexInt, EdgeInt, WeightedGraphInt, GraphMapInt } from './interfaces';
+
+export class Vertex implements VertexInt {
+    value: string;
+
+    constructor(value: string) {
+        this.value = value;
+    }
+}
+
+export class Edge implements EdgeInt {
+    firstVertex: VertexInt;
+    secondVertex: VertexInt;
+    weight: number;
+
+    constructor(vertex1: Vertex, vertex2: Vertex, weight: number) {
+        this.firstVertex = vertex1;
+        this.secondVertex = vertex2;
+        this.weight = weight;
+    }
+}
+
+export class WeightedGraph implements WeightedGraphInt {
+    vertexes: {
+        [key: string]: {
+            [key: string]: number
+        }
+    } = {};
+
+    public addVertex(vertex: Vertex) {
+        if(!this.vertexes[vertex.value]) {
+            this.vertexes[vertex.value] = {};
+        }
+        
+    }
+
+    public addEdge(firstVertex: Vertex, secondVertex: Vertex, weight: number): void | string {
```

instead of returning string, you should probably throw an error